src/main/java/no/sikt/nva/pubchannels/dataporten/DataportenAuthClient.java

```diff
@@ -0,0 +1,83 @@
+package no.sikt.nva.pubchannels.dataporten;
+
+import no.sikt.nva.pubchannels.handler.AuthClient;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadGatewayException;
+import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static no.unit.nva.commons.json.JsonUtils.dtoObjectMapper;
+import static nva.commons.core.attempt.Try.attempt;
+
+public class DataportenAuthClient implements AuthClient {
```

Comment:
Can we use HttpClient::authenticator for this instead?

Reply:
It's a bug in 11. Fixed in 13.

https://stackoverflow.com/questions/54208945/java-11-httpclient-not-sending-basic-authentication